src/R/Support/Impl/Help/Packages/PackageIndex.cs

```diff
@@ -201,38 +200,55 @@ public sealed class PackageIndex : IPackageIndex {
             }
         }
 
+        private async Task UpdateInstalledPackagesAsync() {
+            var token = await _buildIndexLock.WaitAsync();
```

Comment:
ResetAsync makes code think that the index is not ready. Basically it is not set when index is empty and set otherwise. Technically states are 'empty', 'ready' and 'needs update'. Perhaps I should add `finally` then